pkg/auth/authproviders/openshift/backend_impl.go

```diff
@@ -109,13 +102,9 @@ func (b *backend) Config() map[string]string {
 func (b *backend) LoginURL(clientState string, ri *requestinfo.RequestInfo) (string, error) {
 	state := idputil.MakeState(b.id, clientState)
 
-	// baseRedirectURL does not include the hostname, take it from the request.
-	// Allow HTTP only if the client did not use TLS and the host is localhost.
-	redirectURL := b.baseRedirectURL
-	redirectURL.Host = ri.Hostname
-	if !ri.ClientUsedTLS && netutil.IsLocalEndpoint(redirectURL.Host) {
-		redirectURL.Scheme = "http"
-	}
+	// baseRedirectURLPath does not include the hostname, take it from the
+	// request.
```

Comment:
This is now a bit misleading, how about

```suggestion
	// Augment baseRedirectURLPath to a redirect URL with hostname, etc set.
```